packages/block-library/src/query-total/index.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Server-side rendering of the `core/query-total` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `query-total` block on the server.
+ *
+ * @since 6.8.0
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string The rendered block content.
+ */
+function render_block_core_query_total( $attributes, $content, $block ) {
+	global $wp_query;
+	$wrapper_attributes = get_block_wrapper_attributes();
+	if ( isset( $block->context['query']['inherit'] ) && $block->context['query']['inherit'] ) {
+		$query_to_use = $wp_query;
+		$current_page = max( 1, get_query_var( 'paged', 1 ) );
+	} else {
+		$page_key     = isset( $block->context['queryId'] ) ? 'query-' . $block->context['queryId'] . '-page' : 'query-page';
+		$current_page = isset( $_GET[ $page_key ] ) ? (int) $_GET[ $page_key ] : 1;
+		$query_to_use = new WP_Query( build_query_vars_from_query_block( $block, $current_page ) );
+	}
+
+	$max_rows       = $query_to_use->found_posts;
+	$posts_per_page = $query_to_use->get( 'posts_per_page' );
+
+	// Calculate the range of posts being displayed.
+	$start = ( $current_page - 1 ) * $posts_per_page + 1;
+	$end   = min( $start + $posts_per_page - 1, $max_rows );
+
+	// Prepare the display based on the `displayType` attribute.
+	$output = '';
+	switch ( $attributes['displayType'] ) {
+		case 'range-display':
+			if ( $start === $end ) {
+				$range_text = sprintf(
+					/* translators: 1: Start index of posts, 2: Total number of posts */
+					__( 'Displaying %1$s of %2$s' ),
+					'<strong>' . $start . '</strong>',
+					'<strong>' . $max_rows . '</strong>'
+				);
+			} else {
+				$range_text = sprintf(
+					/* translators: 1: Start index of posts, 2: End index of posts, 3: Total number of posts */
+					__( 'Displaying %1$s – %2$s of %3$s' ),
+					'<strong>' . $start . '</strong>',
+					'<strong>' . $end . '</strong>',
+					'<strong>' . $max_rows . '</strong>'
+				);
+			}
+
+			$output = sprintf( '<p>%s</p>', $range_text );
+			break;
+
+		case 'total-results':
+		default:
+			$output = sprintf(
+				'<p><strong>%d</strong> %s</p>',
+				$max_rows,
+				_n( 'result found', 'results found', $max_rows )
```

Comment:
Good one, @swissspidy. Thank you for flagging it. There were a few iterations on translations in this PR as documented in https://github.com/WordPress/gutenberg/pull/67629#discussion_r1876170366. This needs a bit of practice to get it right 😄

Reply:
> This would be better as _n( '<strong>%d</strong> result found',  '<strong>%d</strong> results found', $max_rows )

Considering this https://github.com/WordPress/gutenberg/pull/67629#discussion_r1876278429 I believe it was a better option to use `_n( 'result found', 'results found', $max_rows )`

@swissspidy

Reply:
Well no, HTML is discouraged unless necessary.

Here, HTML is better than string concatenation. So it should be fixed.

Reply:
So after adopting the suggested changes & avoiding `<strong>` tag this is how things are looking.

```php
$output = sprintf(
  '<p>%s</p>',
  $max_rows,
  _n( '%d result found', '%d results found', $max_rows )
);
```

Does this require any changes?

Reply:
looks good to me, apart from the missing translator comment

Reply:
> looks good to me, apart from the missing translator comment

Yes, thats added.

Reply:
Umm there was a logical issue with the previous solution, this is the updated one

```php
$total_text = sprintf( _n( '%d result found', '%d results found', $max_rows ), $max_rows );
$output = sprintf(
  '<p>%s</p>',
  $total_text,
);
```

Reply:
Added these changes through https://github.com/WordPress/gutenberg/pull/68150

Reply:
Can someone review & get this PR (https://github.com/WordPress/gutenberg/pull/68150) merged?